### PR TITLE
Implement SkeletonData, RabbitData and OcelotData

### PR DESCRIPTION
**Pull request by sanman00**
Implementation of `SkeletonData`, `RabbitData` and `OcelotData`.

Implementation of the `DataManipulator`s:
- [x] Registration of field getters and setters
- [x] Accurately used the appropriate `AbstractData` implementation

Implementation of the `ImmutableDataManipulator`s:
- [x] Accurately extended the appropriate `AbstractImmutableData` implementation
- [x] Accurately instantiating final instance fields with an `ImmutableValue` counter part for any value getters
- [x] If necessary, creating a new `ImmutableValue` for a value that should not be cached, or, using the existing caching utils.

Implementation of the `ValueProcessor`s:
- [x] Appropriately extended `AbstractSpongeValueProcessor`
- [x] Individual processors for each source type possible (one for `ItemStack`/`TileEntity`/`Entity` as necessary).

Registration:
- [x] Registered the `Key`s correctly in the `KeyRegistry`
- [x] Registered the `DataProcessor`s and `DataManipulatorBuilder` in `SpongeSerializationRegistry`
- [x] Registered the `ValueProcessor`s in the `SpongeSerializationRegistry`

[Test plugin for all 3 types of data](http://pastebin.com/dgMRyivu)

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/processor/value/entity/SkeletonTypeValueProcessor.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    protected Optional<SkeletonType> getVal(EntitySkeleton container) {
+        return Optional.ofNullable(SpongeEntityConstants.SKELETON_IDMAP.get(container.getSkeletonType()));
+    }
+    
+    @Override
+    protected ImmutableValue<SkeletonType> constructImmutableValue(SkeletonType value) {
+        return ImmutableSpongeValue.cachedOf(Keys.SKELETON_TYPE, SkeletonTypes.NORMAL, value);
+    }
+
+}
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/common/data/processor/value/entity/SkeletonTypeValueProcessor.java]: Missing new line

File: src/main/java/org/spongepowered/common/mixin/core/entity/living/monster/MixinEntitySkeleton.java [hunk 2]
```diff
@@ -25,6 +25,10 @@
 package org.spongepowered.common.mixin.core.entity.living.monster;
 
 import net.minecraft.entity.monster.EntitySkeleton;
+
+import java.util.List;
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/entity/living/monster/MixinEntitySkeleton.java]: Fix your import styling.

**Comment by gabizou** [comment]
@sanman00 If you can, rebase your PR and I'll get to it within the day.

**Comment by sanman00** [comment]
@gabizou Done

**Pull request closed without merging**